File: src/main/java/senkohotel/suzu/command/ImageCommand.java
package senkohotel.suzu.command;

import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
import senkohotel.hotelbot.Main;
import senkohotel.hotelbot.commands.Command;
import senkohotel.hotelbot.utils.MessageUtils;
import senkohotel.suzu.xp.XPCollection;

public class ImageCommand extends Command {
    public ImageCommand() {
        super();
        name = "image";
        desc = "embed perms thing";
    }

    @Override
    public void exec(MessageReceivedEvent msg, String[] args) {
        super.exec(msg, args);

        EmbedBuilder embed = new EmbedBuilder()
                .setDescription(
                        "Hi\nYou might ask 'Why cant i send images?'\nor 'Why is that link not making an image appear?'\nJust chat a bit more until you get the **" +
                                msg.getGuild().getRoleById(XPCollection.roles.get(0).roleID).getName() + "** " + XPCollection.roles.get(0).roleIcon + " role!\n" +
                                "You can check your progress with `suzu rank` in <#843148981939994624>"
                ).setColor(Main.accentColor);
        MessageUtils.send(msg.getChannel().getId(), embed);
        msg.getMessage().delete().complete();
    }
}

File: Suzu/Commands/ImageCommand.cs
using System.Text;
using DSharpPlus;
using DSharpPlus.Entities;
using HotelLib;
using HotelLib.Commands;
using HotelLib.Utils;

namespace Suzu.Commands;

public class ImageCommand : SlashCommand
{
    public override string Name => "image";
    public override string Description => "Shows the image permission embed";
    public override Permissions Permissions => Permissions.ManageMessages;

    public override void Handle(HotelBot bot, DiscordInteraction interaction)
    {
        var content = new StringBuilder();
        content.AppendLine("Hi");
        content.AppendLine("You might ask 'Why cant i send images?'");
        content.AppendLine("or 'Why is that link not making an image appear?'");
        content.AppendLine("Just chat a bit more until you get the **Classic-Membership**<:classicTail:992862741561868388> role!");
        content.AppendLine("You can check your progress with `/rank` in <#843148981939994624>");

        var embed = new DiscordEmbedBuilder()
                    .WithDescription(content.ToString())
                    .WithColor(new DiscordColor(221, 164, 137));

        interaction.Channel.SendMessageAsync(embed);
        interaction.Reply("Message sent!", true);
    }
}
